### ISSUE TITLE
Add a trait for registers

### ISSUE THREAD

**Issue by bjorn3** [headcrab-rs/headcrab#87]
Something like

```rust
trait Registers {
    fn ip(&self) -> u64;
    fn set_ip(&mut self, ip: u64);
    fn sp(&self) -> u64;
    fn set_sp(&mut self, sp: u64);
    fn bp(&self) -> Option<u64>; // None if the standard abi on the platform has no base pointer
    #[must_use]
    fn set_bp(&mut self, bp: u64) -> Option<()>; // None if the standard abi on the platform has no base pointer

    fn reg_for_dwarf(&self, reg: gimli::Register) -> Option<u64>; // None when register doesn't exist
    #[must_use]
    fn set_reg_for_dwarf(&mut self, reg: gimli::Register, val: u64) -> Option<()>; // None when register doesn't exist

    fn name_for_dwarf(reg: gimli::Register) -> Option<&'static str>; // None when register doesn't exist
    fn dwarf_for_name(name: &str) -> Option<gimli::Register>; // None when register doesn't exist
}
```

I am not quite sure how to handle different reg sizes on different architectures and within a single architecture though.

**Comment by nbaksalyar** [comment]
This is a prerequisite for #96. I'd like to give it a try.

**Comment by Stupremee** [comment]
Probably we can represent the different register sizes using an Enum with a u64, u32, ... variants and with a bunch of helper methods and `TryInto` impls it should be easy to get the raw values out of the Enum.

**Comment by nbaksalyar** [comment]
Closed by #98
